src/dense/generic_dense.jl

```diff
@@ -459,25 +649,32 @@ function ode_interpolation!(vals, tvals, id::I, idxs, deriv::D, p,
                 current_alg = id.alg_choice[i₊]
                 @inbounds cache_i₊ = cache.caches[current_alg] # this alloc is costly
             end
-            _ode_addsteps!(ks[i₊], ts[i₋], timeseries[i₋], timeseries[i₊], dt, f, p,
-                cache_i₊) # update the kcurrent
-            if eltype(vals) <: AbstractArray
-                ode_interpolant!(vals[j], Θ, dt, timeseries[i₋], timeseries[i₊], ks[i₊],
-                    cache_i₊, idxs, deriv, differential_vars)
-            else
-                vals[j] = ode_interpolant(Θ, dt, timeseries[i₋], timeseries[i₊], ks[i₊],
-                    cache_i₊, idxs, deriv, differential_vars)
+        elseif cache isa DefaultCache
+            if current_alg != id.alg_choice[i₊] # switched algorithm
+                current_alg = id.alg_choice[i₊]
+                if current_alg == 1
+                    cache_i₊ = cache.cache1
+                elseif current_alg == 2
+                    cache_i₊ = cache.cache2
+                elseif current_alg == 3
+                    cache_i₊ = cache.cache3
+                elseif cerrent_alg == 4
+                    cache_i₊ = cache.cache4
+                elseif current_alg == 5
+                    cache_i₊ = cache.cache5
+                elseif current_alg == 6
+                    cache_i₊ = cache.cache6
+                end
```

Comment:
Is this actually used anywhere?

Reply:
yeah. It's used further down in the function. Note that I removed the `else` so that it now runs for all cache types.

Reply:
Don't you need to use it in the same part of the for loop to avoid the dynamism?

Reply:
I guess that would be better. I was following the design for Composite, but I guess that doesn't make as much sense for default.